src/middlewared/middlewared/plugins/docker/state_setup.py

```diff
@@ -51,11 +52,27 @@ async def validate_fs(self):
 
     @private
     async def validate_interfaces(self):
-        default_iface, success = await self.middleware.run_in_thread(wait_for_default_interface_link_state_up)
-        if default_iface is None:
-            raise CallError('Unable to determine default interface')
-        elif not success:
-            raise CallError(f'Default interface {default_iface!r} is not in active state')
+        # Retry for up to 60 seconds to handle race condition where system.ready
```

Comment:
Can this logic be fixed in `wait_for_default_interface_link_state_up` instead? This change seems to scatter the same logic over two different places.